### PR TITLE
Snapshot serializer and alias fix

### LINKED ISSUE #87: Use the renamed field name for a span if there is one
Given this query

```graphql
{
  myAlias: field
}
```

You currently get a span with the name "field", where it should be "myAlias"

### PR DESCRIPTION
**Pull request by cliedeman**
Closes #87

### CODE REVIEW

File: src/__tests__/integration-test.ts [hunk 1]
```diff
@@ -102,7 +98,7 @@ class MockTracer {
 
     return {
       log(object) {
-        self.spans.find(span => span.name === name).logs.push(object);
+        self.spans.find(span => span.id === spanId).logs.push(object);
```
Comment (cliedeman) [inline src/__tests__/integration-test.ts]: There was a bug in this logic that broke for array tests but went unnoticed in the old snapshots.

File: src/__tests__/__snapshots__/integration-test.ts.snap [hunk 2]
```diff
@@ -1,315 +1,109 @@
...
+   +-- one:3
+         finished: true
+         logs:
+         1. {"result":"1"}
+   +-- two:4
+         finished: true
+         logs:
+         1. {"result":"2"}
+   +-- one:5
+         finished: false
```
Comment (cliedeman) [inline src/__tests__/__snapshots__/integration-test.ts.snap]: Looks like there is a bug when it comes to marking array spans as finished

**Comment by DanielMSchmidt** [comment]
Cool, great job 👍 I would be super happy to merge it in and I am very happy with the changes you made to the serializer, while the previous implementation was a great step forward it's for more readable now.

I really like how you split up the commits, but it would be great if you could reword them (`git rebase -i HEAD~4` should help you with this).
We need to follow the angular commit guidelines for semantic release to do it's job of automatically releasing the right major minor or patch version. For more info see here: https://github.com/semantic-release/semantic-release#commit-message-format

**Comment by cliedeman** [comment]
Rebased

**Comment by DanielMSchmidt** [comment]
🙈 oh you squashed them. Well, it's fine like this, but could you explain why the scope is `pencil`? Am I missing something? Otherwise you could also use a message without a scope: `feat: implement aliases`

**Comment by cliedeman** [comment]
I didn't want to squash it when I started... and then I lost the original branch :disappointed:

**Comment by cliedeman** [comment]
Phew ok. commits fixed

**Comment by DanielMSchmidt** [comment]
If I wanted to nit-pick the span serializer would normally considered `test` as it only touches tests, but let's get it in like this, the effect should be the very same for our version number ;)

**Comment by cliedeman** [comment]
I actually also use semantic-release on internal projects but shied away from using it on opensource projects because I want to keep the barrier to contribute low. Will be good to see how it works here

**Comment by DanielMSchmidt** [comment]
@cliedeman If I would have seen you as a beginner I would have just squashed the commit for you and reworded it, that way I can still keep the barrier relatively low ;)

**Pull request merged**